angular-c3.js

```diff
@@ -5,27 +5,139 @@ angular.module('c3', [])
 
 .directive('c3', function() {
   console.log('loading directive');
+
+  function getUnload(prev, curr) {
+    var prevKeys;
+    if (prev && prev.columns && prev.columns.length) {
+      prevKeys = {};
+      prev.columns.forEach(function(data) {
+        prevKeys[data[0]] = true;
+      });
+    } else if (prev && prev.rows && prev.rows.length) {
+      prevKeys = {};
+      prev.rows[0].forEach(function(key) {
+        prevKeys[key] = true;
+      });
+    } else {
+      return false;
+    }
+
+    var currKeys;
+    if (curr && curr.columns && curr.columns.length) {
```

Comment:
Will need to remove the code duplication between the `prevKeys` & `currKeys` code.